v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/PipelineController.java

```diff
@@ -184,6 +185,10 @@ private static void setupLogicalDbMigration(
           configContainer.getSrcTableToShardIdColumnMap(
               tableSelector.getSchemaMapper(), spannerTables);
 
```

Comment:
Can we add a top level check to not get into the whole injection code if the pipeline parameter is not set?

The code has many layers, but would this lead to spurious logs of
```
 if (jsonParameter == null || jsonParameter.trim().isEmpty()) {
      LOG.error("JSON parameter for ErrorInjectionPolicyFactory cannot be null or empty.");
      return new NoOpPolicy();
    }

```

Also we probably don't need a NOOP policy if we can entirely avoid the injection frameworks at the root of the pipeline controller.


It might be a bit less intrusive to inject the interceptor only for test environments and avoid adding a user visible parameter - though I am not sure how easy/difficult that would be.